python/sqlalchemy/src/yuga_demo.py

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker, relationship
+from http.server import HTTPServer, BaseHTTPRequestHandler
+import json
+from io import BytesIO
+import jsonpickle
+from yuga_demo_model import User, Order, Product, OrderLine
+
+## ######################################################
+## Configure the parameters below to connect to Yugabyte database: 
+## 
+listen_port = 8000
+db_user = 'postgres'
+db_password = 'pwd'
+database = 'alchemy_orm_sample'
+db_host = 'localhost'
+## Yugabyte
+db_port = 5433
+## Postgresql
+#db_port = 5432
+
+## #####################################################
+
+# ## SQLAlchemy ORM to database
+# With SQLAlchemy we can deal with objects, rather than dealing with rows and columns
+# 
+# Set up the session object, then the session communicates with the database.  Objects are "mapped" to the database tables using annotations (__tablename__), and by extending the Base object.
+
+from sqlalchemy import create_engine
+#dialect+driver://username:password@host:port/database
+engine = create_engine('postgresql://%s:%s@%s:%s/%s' % 
+                       (db_user, db_password, db_host, db_port, database), echo=True)
+Session = sessionmaker(bind=engine)
+# Base = declarative_base()
+
+
+def add_object(session, to_add):
+    session.add(to_add)
+    print('Added: %s' % str(to_add) )
+
+
+def add_objects(session, object_list):
+    session.add_all(object_list)
+    print('Added (list): ')
+    [print(str(obj)) for obj in object_list]
+
+class UserOrdersDAO:
```

Comment:
I feel this class is wrongly named... we are adding methods like `list_products()` here which has nothing to do with a `UserOrdersDAO`. Can we move some of these methods directly into the REST handler functions?

Reply:
If you feel strongly about this, I will change it.  It is my opinion that putting the actual data access methods into a separate class is cleaner than adding this logic into each handler.  One could reason that if you wanted to change this to use a different db, then you could just swap out the DAO.

I have named it more generically so that it is appropriate for users, products and orders (and order lines).

Reply:
I am ok with a DAO, just that the name `UserOrdersDAO` does not have `Products` in it, so becomes confusing. Would rather this be named something like `YugaByteDAO` so that it captures the fact that it is a data access abstraction layer for YugaByte DB.

Reply:
Changed to YugabyteDAO.  Thank you, I think that is better.